scalafix/rules/src/main/scala/utils/CompilerService.scala

```diff
@@ -38,15 +39,43 @@ object CompilerService {
     }
   }
 
-  def getContext(name: Token, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] =
-    getContext(name.pos.start, g)
+  def getContext(name: Term, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] = {
+    val gpos = unit.position(name.pos.start)
+    getContext(gpos, g)
+  }
+
+  def getContext(tree: Tree, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] = {
+    val gpos = getExactPos(tree.pos, g)
+    getContext(gpos, g)
+  }
 
-  def getContext(name: Term.Name, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] =
-    getContext(name.pos.start, g)
+  def getGlobalTree(tree: Tree, g: Global)(implicit unit: g.CompilationUnit): Option[g.Tree] = {
+    val gpos  = getPosAfter(tree.pos, g)
+    val gtree = Try(GlobalProxyService.typedTreeAt(g, gpos)).toOption
+    if (gtree.isDefined && gtree.get.isInstanceOf[g.Template]) {
+      val gpos = getPosBefore(tree.pos, g)
+      Try(GlobalProxyService.typedTreeAt(g, gpos)).toOption
+    } else gtree
+  }
 
-  private def getContext(position: Int, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] = {
-    val gpos  = unit.position(position)
+  private def getContext(gpos: scala.reflect.internal.util.Position, g: Global): Option[g.Context] = {
     val gtree = GlobalProxyService.typedTreeAt(g, gpos)
     Try(g.doLocateContext(gtree.pos)).toOption
   }
+
+  private def getPosAfter(scalaMetaPos: Position, g: Global)(implicit
+    unit: g.CompilationUnit
+  ): scala.reflect.internal.util.Position =
```

Comment:
An alias of this type can make the code easier to read:

```
import scala.reflect.internal.util.{Position => ReflectPos}
```

Reply:
ok